keras_cv/layers/preprocessing/equalize.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+
+class Equalize(tf.keras.layers.Layer):
+    """Equalize performs histogram equalization on a channel-wise basis."""
+
+    def __init__(self, rate=1.0, **kwargs):
+        super().__init__(**kwargs)
+
+    @staticmethod
+    def scale_channel(image, c):
+        image = tf.cast(image[..., c], tf.int32)
```

Comment:
This is assuming channel last format. Is it safe to have this assumption?

Reply:
Also the cast seems interesting here. This is expecting the image to be in a certain range? eg [0, 255]? Might want to put this in the docstring.